source/Mlos.Python/mlos/Optimizers/ExperimentDesigner/UtilityFunctionOptimizers/UtilityFunctionOptimizer.py

```diff
@@ -28,6 +28,13 @@ def __init__(
         self.optimization_problem = optimization_problem
         self.utility_function = utility_function
 
-    @abstractmethod
     def suggest(self, context_values_dataframe: pd.DataFrame) -> Point:
-        raise NotImplementedError
+        config_to_suggest = self.maximize(lambda features: self.utility_function(features).utility,
+                                          context_values_dataframe=context_values_dataframe)
+        self.logger.debug(f"Suggesting: {str(config_to_suggest)}")
+        return config_to_suggest
+
+    @abstractmethod
+    def maximize(self, target_function, context_values_dataframe=None):

```

Comment:
why?

Reply:
Sorry, what is the question? Why is this abstract?

Reply:
Sorry, I didn't understand why we needed the `maximize` function in the first place. After our call, I now understand.

For the benfit of other reviewers:

It's to use the UtilityFunctionOptimizer to optimize the predicted value as well as the utility function (for example to compute a speculative optimum).

Also, per our conversation, I think one way to achieve that goal is to instantiate a separate UtilityFunctionOptimizer for that purpose. This would allow us to keep the UtilityFunctionOptimizers stateful. 

Stateful optimizers can remember "good" points from previous invocations and use them to jump-start the next invocation (under the assumption that the surrogate model changes only a little between invocations).